File: annotation-processor/src/main/java/io/quarkiverse/jimmer/graphql/apt/JimmerGraphQLSourceMethod.java
package io.quarkiverse.jimmer.graphql.apt;

import java.util.List;

import com.squareup.javapoet.AnnotationSpec;

record JimmerGraphQLSourceMethod(
        String name,
        String rawAccessorName,
        String returnType,
        boolean collection,
        String elementType,
        boolean complex,
        boolean transientResolver,
        String graphQLName,
        List<AnnotationSpec> annotations) {
}

File: ksp-processor/src/main/kotlin/io/quarkiverse/jimmer/graphql/ksp/JimmerGraphQLSourceMethod.kt
package io.quarkiverse.jimmer.graphql.ksp

data class JimmerGraphQLSourceMethod(
    val name: String,
    val rawAccessorName: String,
    val returnType: String,
    val collection: Boolean,
    val elementType: String,
    val complex: Boolean,
    val transientResolver: Boolean,
    val graphQLName: String,
    val annotations: List<JimmerGraphQLSourceAnnotation>,
)
